pkg/providers/instancetype/suite_test.go

```diff
@@ -146,7 +146,37 @@ var _ = Describe("InstanceType Provider", func() {
 		ExpectCleanedUp(ctx, env.Client)
 	})
 
-	Context("subscription level quota error responses", func() {
+	Context("vm creation error responses", func() {
+
+		It("should delete the network interface on failure to create the vm", func() {
+			genericAzureErrMsg := "some generic error message"
+			genericSadErrorCode := "GenericSadCase"
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+			azureEnv.VirtualMachinesAPI.VirtualMachinesBehavior.VirtualMachineCreateOrUpdateBehavior.Error.Set(
+				&azcore.ResponseError{
+					ErrorCode: genericSadErrorCode,
+					RawResponse: &http.Response{
+						Body: createSDKErrorBody(genericSadErrorCode, genericAzureErrMsg),
+					},
+				},
+			)
+			pod := coretest.UnschedulablePod()
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+			ExpectNotScheduled(ctx, env.Client, pod)
+
+			// We should have created a nic for the vm
+			Expect(azureEnv.NetworkInterfacesAPI.NetworkInterfacesCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+			// The nic we used in the vm create, should be cleaned up if the vm call fails
+			nic := azureEnv.NetworkInterfacesAPI.NetworkInterfacesCreateOrUpdateBehavior.CalledWithInput.Pop()
```

Comment:
This should be the `DeleteBehavior`, right?

Reply:
Or I guess do we want to check that the DeleteBehavior is called at all? Or just rely on the ok == false check from Load?

Reply:
We should test both delete and create. These tests were added before, will update them.